merlin/analysis/filterbarcodes.py

```diff
@@ -57,118 +58,301 @@ def _run_analysis(self, fragmentIndex):
         barcodeDB.write_barcodes(currentBC, fov=fragmentIndex)
 
 
-class AdaptiveFilterBarcodes(analysistask.AnalysisTask):
+class GenerateAdaptiveThreshold(analysistask.AnalysisTask):
 
     """
-    An analysis task that filters barcodes based on a mean intensity threshold
-    for each area based on the abundance of blank barcodes. The threshold
-    is selected to achieve a specified misidentification rate.
+    An analysis task that generates a three-dimension mean intenisty,
+    area, minimum distance histogram for barcodes as they are decoded.
     """
 
     def __init__(self, dataSet, parameters=None, analysisName=None):
         super().__init__(dataSet, parameters, analysisName)
 
-        if 'misidentification_rate' not in self.parameters:
-            self.parameters['misidentification_rate'] = 0.05
+        # ensure decode_task is specified
+        decodeTask = self.parameters['decode_task']
 
     def fragment_count(self):
         return len(self.dataSet.get_fovs())
 
-    def get_barcode_database(self):
-        return barcodedb.PyTablesBarcodeDB(self.dataSet, self)
-
     def get_estimated_memory(self):
-        return 1000
+        return 5000
 
     def get_estimated_time(self):
-        return 30
+        return 1800
 
     def get_dependencies(self):
-        return [self.parameters['decode_task']]
```

Comment:
Decode task should still be listed as a dependency.

Reply:
Decode task isn't listed as a dependency here so that the GenerateAdaptiveThreshold task can begin at the same time as the decode task and the adaptive threshold can be updated as new fovs are decoded rather than waiting for all decoding to complete. If the decode task is listed as a dependency then GenerateAdaptiveThreshold won't start until all decoding is complete.